x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/steps/blocks/action/edit_step_description_modal.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useState } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+  EuiText,
+  EuiTextArea,
+  useEuiTheme,
+} from '@elastic/eui';
+import { css } from '@emotion/react';
+import { i18n } from '@kbn/i18n';
+import type { StreamlangProcessorDefinitionWithUIAttributes } from '@kbn/streamlang';
+import { getStepDescription } from './utils';
+
+export interface EditStepDescriptionModalProps {
+  step: StreamlangProcessorDefinitionWithUIAttributes;
+  mode: 'add' | 'edit';
+  onSave: (description: string) => void;
+  onCancel: () => void;
+}
+
+export const EditStepDescriptionModal: React.FC<EditStepDescriptionModalProps> = ({
+  step,
+  mode,
+  onSave,
+  onCancel,
+}) => {
+  const { euiTheme } = useEuiTheme();
+  const modalHeaderCss = css`
+    padding-block-end: ${euiTheme.size.xs};
+  `;
+
+  const initialValue = useMemo(() => {
+    if (step.description && step.description.trim().length > 0) {
+      return step.description;
+    }
+
+    return getStepDescription({
+      ...step,
+      description: undefined,
+    });
+  }, [step]);
+
+  const [value, setValue] = useState(initialValue);
+
+  const handleSave = () => {
+    onSave(value.trim());
+  };
+
+  const isEditMode = mode === 'edit';
+
+  const title = isEditMode
+    ? i18n.translate('xpack.streams.enrichment.processor.editDescription.title', {
+        defaultMessage: 'Edit description',
+      })
+    : i18n.translate('xpack.streams.enrichment.processor.addDescription.title', {
+        defaultMessage: 'Add description',
+      });
```

Comment:
In a few other components I saw we're moving these translations into constants, IMO this make the component's code a bit cleaner and easier to read.